src/app/layout.tsx

```diff
@@ -1,17 +1,52 @@
 import { getBaseURL } from "@lib/util/env"
+import { getI18NConfigCallback } from "@lib/i18n/config-callback"
+import { LOCALE_COOKIE, fallbackLng, languages } from "@lib/i18n/settings"
 import { Metadata } from "next"
+import { NextIntlClientProvider } from "next-intl"
+import { setRequestLocale } from "next-intl/server"
+import { cookies } from "next/headers"
+import { Suspense } from "react"
 import "styles/globals.css"
 
 export const metadata: Metadata = {
   metadataBase: new URL(getBaseURL()),
+};
+
+export function generateStaticParams() {
+  return languages.map((locale) => ({ locale }));
 }
 
-export default function RootLayout(props: { children: React.ReactNode }) {
+export default async function RootLayout({
+  children,
+  params: { locale: requestedLocale },
+}: {
+  children: React.ReactNode;
+  params: { locale: string };
+}) {
+  // determine the locale to use based on URL, cookie, or fallback
+  const isValidLocale = languages.includes(requestedLocale);
+  const cookieLocale = (await cookies()).get(LOCALE_COOKIE)?.value;
+  const finalLocale = isValidLocale 
+    ? requestedLocale 
+    : cookieLocale && languages.includes(cookieLocale)
+      ? cookieLocale
+      : fallbackLng;
```

Comment:
You don't need to negotiate a locale yourself, this is done for you in the middleware.